Tuwat crashes on endpoint unavailable

```
{"level":"info","ts":1708430586.682414,"caller":"aggregation/aggregator.go:201","msg":"Collected alerts","collector":"Alertmanager (https://alertmanager.system.dev.contargo.cloud)","tag":"dev","count":0,"error":"error delivering result send on closed channel"}
panic: close of closed channel
goroutine 19037 [running]:
github.com/synyx/tuwat/pkg/aggregation.(*Aggregator).collect(0xc0001f4420, {0xcfbe20, 0xc00018a0c0}, 0xc000657800)tuwat/pkg/aggregation/aggregator.go:228 +0x608
created by github.com/synyx/tuwat/pkg/aggregation.(*Aggregator).Run in goroutine 15
tuwat/pkg/aggregation/aggregator.go:161 +0x5a9
```

--- a/pkg/aggregation/aggregator.go
+++ b/pkg/aggregation/aggregator.go
@@ -167,6 +167,9 @@ func (a *Aggregator) Run(ctx context.Context) {
 }
 
 func (a *Aggregator) collectAggregate(ctx context.Context, collect <-chan result) {
+	ctx, cancel := context.WithTimeout(ctx, a.interval)
+	defer cancel()
+
 	var results []result
 outer:
 	for {

--- a/pkg/aggregation/aggregator.go
+++ b/pkg/aggregation/aggregator.go
@@ -136,11 +136,10 @@ func (a *Aggregator) Run(ctx context.Context) {
 	ticker := a.clock.Ticker(a.interval)
 	defer ticker.Stop()
 
-	collect := make(chan result, 20)
-	var results []result
-
 	otelzap.Ctx(ctx).Info("Collecting on Start")
+	collect := make(chan result, 20)
 	go a.collect(ctx, collect)
+	go a.collectAggregate(ctx, collect)
 
 	active := true
 	for {
@@ -158,20 +157,33 @@ func (a *Aggregator) Run(ctx context.Context) {
 				active = true
 			}
 
+			collect := make(chan result, 20)
 			go a.collect(ctx, collect)
-		case r, ok := <-collect:
-			if !ok {
-				for _, dashboard := range a.dashboards {
-					a.aggregate(ctx, dashboard, results)
-				}
-				results = nil
-				collect = make(chan result, 20)
-			} else if ok {
-				results = append(results, r)
-			}
+			go a.collectAggregate(ctx, collect)
 		case <-ctx.Done():
 			return
 		}
+	}
+}
+
+func (a *Aggregator) collectAggregate(ctx context.Context, collect <-chan result) {
+	var results []result
+outer:
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case r, ok := <-collect:
+			if !ok {
+				break outer
+			}
+
+			results = append(results, r)
+		}
+	}
+
+	for _, dashboard := range a.dashboards {
+		a.aggregate(ctx, dashboard, results)
 	}
 }
 

